projects/cli/src/effects.js

```diff
@@ -1,622 +1,725 @@
-      Object.values(Utils.customizableFiles)
-        .flatMap(({ filepaths }) => filepaths.map(filepath => path.join(process.cwd(), 'src', ...filepath.split('/'))))
-    let customizedFilepaths = chokidar.watch(customizableFileFilepaths, {
-      ignorePermissionErrors: true,
-      ignoreInitial: true
-    })
-    customizedFilepaths.on('all', syncCustomizableFiles)
-
-    // Check config for Elm debugger options
-    let debug = false
-    try { debug = config.app.elm[mode()].debugger }
-    catch (_) { }
-
-    // Run the vite server on options.port
-    server = await Vite.createServer({
-      configFile: false,
-      root: path.join(process.cwd(), '.elm-land', 'server'),
-      publicDir: path.join(process.cwd(), 'static'),
-      envDir: process.cwd(),
-      envPrefix: 'ELM_LAND_',
-      server: {
-        host: options.host,
-        port: options.port,
-        fs: { allow: ['../..'] }
-      },
-      plugins: [
-        DotPathFixPlugin.plugin(),
-        ElmVitePlugin.plugin({
-          debug,
-          optimize: false
-        }),
-        TypeScriptPlugin.plugin()
-      ],
-      logLevel: 'silent'
-    })
-
-    server.ws.on('error', (e) => console.error(e))
-    server.ws.on('elm:client-ready', () => {
-      if (lastErrorSent) {
-        server.ws.send('elm:error', {
-          error: ElmErrorJson.toColoredHtmlOutput(lastErrorSent)
-        })
-      }
-    })
-
-    await generateElmFiles(config, server)
-
-    await server.listen()
-
-    return { problem: null, port: server.httpServer.address().port }
-  } catch (e) {
-    console.error(e)
-    console.log('')
-    return { problem: `❗️ Had trouble starting the server...` }
-  }
-}
-
-let lastErrorSent = undefined
+	let server;
+
+	try {
+		let rawConfig = await Files.readFromUserFolder("elm-land.json");
+		let config = JSON.parse(rawConfig);
+
+		// Handle any missing '.elm-land' files
+		await handleElmLandFiles();
+
+		// Expose ENV variables explicitly allowed by the user
+		handleEnvironmentVariables({ config });
+
+		// Listen for changes to the "src" folder, so the browser
+		// automatically refreshes when an Elm file is changed
+		let srcFolder = `${path.join(process.cwd(), "src")}/**/*.elm`;
+		let srcFolderWatcher = chokidar.watch(srcFolder, {
+			ignorePermissionErrors: true,
+			ignoreInitial: true,
+		});
+		let mainElmPath = path.join(process.cwd(), ".elm-land", "src", "Main.elm");
+
+		srcFolderWatcher.on("all", () => {
+			Files.touch(mainElmPath);
+		});
+
+		// Listen for changes to static assets, so the browser
+		// automatically shows the latest asset changes
+		let staticFolder = `${path.join(process.cwd(), "static")}/**`;
+		let staticFolderWatcher = chokidar.watch(staticFolder, {
+			ignorePermissionErrors: true,
+			ignoreInitial: true,
+		});
+		let indexHtmlPath = path.join(
+			process.cwd(),
+			".elm-land",
+			"server",
+			"index.html",
+		);
+
+		// staticFolderWatcher.on("all", () => {
+		// 	Files.touch(indexHtmlPath);
+		// });
```

Comment:
All I did here was comment this out to stop trigger reloads while I test the elm-watch interactions.